contracts/token/ERC7984/ERC7984.sol

```diff
@@ -204,23 +203,29 @@ abstract contract ERC7984 is IERC7984 {
 
         bytes32[] memory cts = new bytes32[](1);
         cts[0] = euint64.unwrap(encryptedAmount);
-        uint256 requestID = FHE.requestDecryption(cts, this.finalizeDiscloseEncryptedAmount.selector);
-        _requestHandles[requestID] = encryptedAmount;
+        FHE.requestDecryption(cts, this.finalizeDiscloseEncryptedAmount.selector);
     }
 
-    /// @dev Finalizes a disclose encrypted amount request.
+    /**
+     * @dev Finalizes a disclose encrypted amount request.
+     * For gas saving purposes, the `requestId` might not be related to the expected
+     * {discloseEncryptedAmount} request. As a result, the current {finalizeDiscloseEncryptedAmount}
+     * function might emit a disclosed amount related to another decryption request context.
+     * In this case it wouldn't reveal private information, but would only display public information
+     * since the handle would have already been allowed for public decryption anyway through a previous
```

Comment:
```suggestion
     * In this case it would only display public information
     * since the handle would have already been allowed for public decryption through a previous
```